### PR TITLE
feat: read variables from other sources when using files

### PR DESCRIPTION
**Pull request by bryan-stclab**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] DevOps process (non-breaking change which improves efficiency and reliability for CI/CD)
- [ ] Documentation only

## What this PR does / why we need it:

* Check the issue #164 for details.

## Which issue/s this PR fixes

#164

### CODE REVIEW

File: core/data-layer/src/data_layer.rs [hunk 1]
```diff
@@ -296,13 +297,23 @@ impl DataLayer {
             return Err(anyhow!(result.err().unwrap().to_string()));
         }
         let result = result.unwrap();
+
+        let config_mapper_data = get_config_mapper();
+
         for row in result {
+            let metadata_str: Option<&str> = row.try_get("metadata").ok();
+            let metadata = if let Some(meta_str) = metadata_str {
```
Comment (pueding) [inline core/data-layer/src/data_layer.rs:305]: instead of "let = if let", how about "let = match"

File: core/data-layer/src/data_layer.rs [hunk 2]
```diff
@@ -296,13 +297,23 @@ impl DataLayer {
...
+                execute_config_mapper(meta_str.to_string(), config_mapper_data.clone()).unwrap()
```
Comment (pueding) [inline core/data-layer/src/data_layer.rs:306]: do not unwrap right away. Add an error handling

Comment (bryan-stclab) [reply core/data-layer/src/data_layer.rs]: I modified the statements to use map_err and ? operator.

File: core/data-layer/src/data_layer.rs (hunk 1 above)
Comment (bryan-stclab) [reply core/data-layer/src/data_layer.rs]: I modified the statements as follows.
```
let metadata = match row.try_get::<Option<&str>, _>("metadata") {
                Ok(Some(metadata_str)) => {
                    execute_variable_mapper(metadata_str.to_string(), variable_mapper_data.clone())
                        .map_err(|e| anyhow!("Error in execute_variable_mapper: {}", e))?
                }
                Ok(None) => serde_json::Value::Null.to_string(),
                Err(e) => return Err(anyhow!("Error getting metadata: {}", e)),
            };
```

**Review by pueding — APPROVED** [review]

**Pull request merged**